pkg/cmd/flagtypes/addr.go

```diff
@@ -84,7 +85,10 @@ func (a *Addr) Set(value string) error {
 		a.Host = host
 		a.Port = int(portNum)
 	} else {
-		port := a.DefaultPort
+		port := 0
+		if !isURL {
```

Comment:
Because you specified https://hostname. Not https://hostname:8080.  You only get default port if you say "0.0.0.0" or "localhost".  https implies port 443 as part of the definition of a URL.

----- Original Message -----
> > @@ -84,7 +85,10 @@ func (a *Addr) Set(value string) error {
> >  		a.Host = host
> >  		a.Port = int(portNum)
> >  	} else {
> > -		port := a.DefaultPort
> > +		port := 0
> > +		if !isURL {
> 
> This means that if I specify `https://hostname` and the defaultPort is
> `8080`, I will get `https://hostname` instead of `https://hostname:8080`?
> Why would I want that to happen?
> 
> ---
> Reply to this email directly or view it on GitHub:
> https://github.com/openshift/origin/pull/2243/files#r30347682